great_expectations/data_context/store/checkpoint_store.py

```diff
@@ -131,12 +131,20 @@ def delete_checkpoint(
                 message=f'Non-existent Checkpoint configuration named "{key.configuration_key}".\n\nDetails: {exc_ik}'  # type: ignore[union-attr]
             )
 
-    def get_checkpoint(self, name: str | None, id: str | None) -> CheckpointConfig:
-        key: Union[GXCloudIdentifier, ConfigurationIdentifier] = self.determine_key(
-            name=name, ge_cloud_id=id
-        )
+    def get_checkpoint(
+        self, name: Optional[ConfigurationIdentifier | str], id: Optional[str]
+    ) -> CheckpointConfig:
```

Comment:
I think this is fine to do as a followup but this seems like we should have 2 overloads defined here.
1 where a `name` is provided and another where `id` is provided.

Reply:
I looked into this, but it turned into a rabbit hole. I think a bunch of the related methods should be refactored before looking into the overloads.